src/libcore/option.rs

```diff
@@ -913,6 +913,65 @@ impl<T> Option<T> {
     pub fn replace(&mut self, value: T) -> Option<T> {
         mem::replace(self, Some(value))
     }
+
+    /// Zips `self` with another `Option`.
+    ///
+    /// Returns `Some((_, _))` when both `self` and `other`
+    /// are `Some(_)`, otherwise return `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(option_zip)]
+    /// let x = Some(1);
+    /// let y = Some("hi");
+    /// let z = None::<u8>;
+    ///
+    /// assert_eq!(x.zip(y), Some((1, "hi")));
+    /// assert_eq!(x.zip(z), None);
+    /// ```
+    #[inline]
```

Comment:
Please remove the `#[inline]` attributes for now. For example, see [this discussion](https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/When.20to.20add.20.23.5Binline.5D.3F) on why it's not a great idea to put `#[inline]` on everything. I know that std currently basically puts `#[inline]` on everything, but I'd like to kind of change this. We can of course add it back if we have benchmark evidence that it helps.